src/layers/src/Pane.js

```diff
@@ -136,64 +84,105 @@ class Pane extends PureComponent {
     return borderSideProperty
   }
 
-  render() {
-    const {
-      theme,
-
-      background,
-
-      elevation,
-      hoverElevation,
-      activeElevation,
-
-      border,
-      borderTop,
-      borderRight,
-      borderBottom,
-      borderLeft,
-
-      css = {},
-      ...props
-    } = this.props
-
-    const elevationStyle = theme.getElevation(elevation)
-    const hoverElevationStyle = this.getHoverElevationStyle(hoverElevation, css)
-    const activeElevationStyle = this.getActiveElevationStyle(
-      activeElevation,
-      css
-    )
-
-    const [_borderTop, _borderRight, _borderBottom, _borderLeft] = [
-      borderTop,
-      borderRight,
-      borderBottom,
-      borderLeft
-    ].map(borderSideProperty =>
-      this.getBorderSideProperty({ borderSideProperty, border })
-    )
-
-    const className = cx(
-      props.className,
-      glamorCss({
-        ...css,
-        ...hoverElevationStyle,
-        ...activeElevationStyle
-      }).toString()
-    )
-
-    return (
-      <Box
-        borderTop={_borderTop}
-        borderRight={_borderRight}
-        borderBottom={_borderBottom}
-        borderLeft={_borderLeft}
-        boxShadow={elevationStyle}
-        background={theme.getBackground(background)}
-        {...props}
-        className={className}
-      />
-    )
-  }
+  const elevationStyle = theme.getElevation(elevation)
+  const hoverElevationStyle = getHoverElevationStyle(hoverElevation, css)
+  const activeElevationStyle = getActiveElevationStyle(
+    activeElevation,
+    css
+  )
+
+  const [_borderTop, _borderRight, _borderBottom, _borderLeft] = [
+    borderTop,
+    borderRight,
+    borderBottom,
+    borderLeft
+  ].map(borderSideProperty =>
+    getBorderSideProperty({ borderSideProperty, border })
+  )
+
+  const className = cx(
+    props.className,
+    glamorCss({
+      ...css,
+      ...hoverElevationStyle,
+      ...activeElevationStyle
+    }).toString()
+  )
+
+  return (
+    <Box
+      innerRef={ref}
+      borderTop={_borderTop}
+      borderRight={_borderRight}
+      borderBottom={_borderBottom}
+      borderLeft={_borderLeft}
+      boxShadow={elevationStyle}
+      background={theme.getBackground(background)}
+      {...props}
+      className={className}
+    />
+  )
+}
+
+Pane.propTypes = {
```

Comment:
Apparently we have to add propTypes to the final resulting exported component instead.

Reply:
So like

```
const MemoizedPane = memo(forwardRef(Pane))

MemoizedPane.propTypes = {...}
```
?

Reply:
exactly!

Reply:
Gettung some proptype list issues when i tried this for the spinner? Do I need to do it on the component itself too?
<img width="519" alt="Screen Shot 2020-04-20 at 21 57 26" src="https://user-images.githubusercontent.com/13311268/79827126-65ff2b80-8352-11ea-87f6-b7006068ee59.png">

Reply:
Can we do something like this instead:

```js
const Spinner = memo(forwardRef((props, ref) => {
  ...
})

Spinner.propTypes = {
  ...
}

export default Spinner
```

Reply:
@mshwery Done, my only gripe is it looks a little messy at the declaration point, but not much we can do I guess. Unless maybe:

```
const Spinner = memo(forwardRef((props, ref) => {
  const {
     ...
  } = props
})

Spinner.propTypes = {
  ...
}

export default Spinner
```

instead of restructuring in the parameters?

Reply:
Yeah would be +1 to destructuring the props in the function body instead of in the function declaration.